tests/integration/types_test.ts

```diff
@@ -0,0 +1,303 @@
+  const customType = DefineType({
+    title: "Title",
+    description: "Description",
+    name: "Name",
+    type: Schema.types.boolean,
+  });
+
+  type Actual = RuntimeType<typeof customType>;
+
+  assert<CanBe<boolean, Actual>>(true);
+});
+
+Deno.test("RuntimeType should provide a usable type from DefineProperty returned object", () => {
+  const testProperty = DefineProperty({
+    title: "test_property",
+    type: Schema.types.object,
+    properties: {},
+    required: [],
+  });
+
+  type Actual = RuntimeType<typeof testProperty>;
+
+  const expected = {};
+
+  assert<CanBe<typeof expected, Actual>>(true);
+});
+
+Deno.test("RuntimeType should provide a usable type from DefineType return object with type: 'object'", () => {
+  const customType = DefineType({
+    title: "Title",
+    description: "Description",
+    name: "Name",
+    type: Schema.types.object,
+    properties: {
+      bool: {
+        type: Schema.types.boolean,
+      },
+      int: {
+        type: Schema.types.integer,
+      },
+      num: {
+        type: Schema.types.number,
+      },
+      string: {
+        type: Schema.types.string,
+      },
+      arr: {
+        type: Schema.types.array,
+        items: {
+          type: Schema.types.boolean,
+        },
+      },
+      obj: DefineProperty({
+        type: Schema.types.object,
+        description: "hello",
+        properties: {
+          obj: DefineProperty({
+            type: Schema.types.object,
+            properties: {
+              obj: DefineProperty({
+                type: Schema.types.object,
+                properties: {
+                  obj: DefineProperty({
+                    type: Schema.types.object,
+                    properties: {
+                      bool: {
+                        type: Schema.types.boolean,
+                      },
+                    },
+                    required: ["obj"],
+                  }),
+                },
+                required: ["obj"],
+              }),
+            },
+            required: ["obj"],
+          }),
+        },
+        required: ["obj"],
+      }),
+    },
+    required: ["bool", "int", "num", "string", "arr", "obj"],
+  });
+
+  type Actual = RuntimeType<typeof customType>;
+
+  const expected = {
+    bool: true,
+    int: 0,
+    num: 0.1,
+    string: "",
+    arr: [true],
+    obj: {
+      obj: {
+        obj: {
+          obj: {
+            bool: true,
+          },
+        },
+      },
+    },
+  };
+
+  assert<CanBe<typeof expected, Actual>>(true);
+});
+
+Deno.test("RuntimeType should provide a usable type from DefineType return object with type: 'object' and empty", () => {
+  const customType = DefineType({
+    title: "Title",
+    description: "Description",
+    name: "Name",
+    type: Schema.types.object,
+    properties: {},
+    required: [],
+  });
+
+  type Actual = RuntimeType<typeof customType>;
+
+  const expected = {};
```

Comment:
I'm not sure what the implied type of an empty object is - this one might be tricky!

But, I think most of your tests will require feeding these custom types / defined properties through a function definition, which then we have to pump through the test utilities we have to create a function handler, invoke the handler, and look at how the input and/or output types compare.

Reply:
Replaced `{}` with `Record<string | number | symbol, never>` where ever possible